sdk/Sdk/Tasks/GenerateFunctionMetadata.cs

```diff
@@ -45,12 +44,16 @@ public override bool Execute()
             {
                 var functionGenerator = new FunctionMetadataGenerator(MSBuildLogger);
 
-                var functions = functionGenerator.GenerateFunctionMetadata(AssemblyPath!, ReferencePaths ?? Enumerable.Empty<ITaskItem>());
+                IEnumerable<SdkFunctionMetadata> functions = functionGenerator.GenerateFunctionMetadata(AssemblyPath!, ReferencePaths ?? Enumerable.Empty<ITaskItem>());
+                IDictionary<string, string> extensions = functionGenerator.Extensions;
 
-                var extensions = functionGenerator.Extensions;
-                var extensionsCsProjGenerator = new ExtensionsCsprojGenerator(extensions, ExtensionsCsProjFilePath!, AzureFunctionsVersion!, TargetFrameworkIdentifier!, TargetFrameworkVersion!);
+                if (!string.IsNullOrEmpty(ExtensionsCsProjFilePath))
+                {
+                    // Null/empty ExtensionsCsProjFilePath means the extension project is externally provided.
```

Comment:
So if it's externally provided, we won't generate? Wondering what it is that makes it not possible for us to run generate for those cases? Is it that we don't know anything about this externally provided project?

Reply:
See the description of this PR. Externally supplied means the user has taken control of the worker extension csproj and they are responsible for ensuring it is 'correct' and building it. They then only supply the path to the csproj to our targets, where we will collect the build output (without building).

When the project is externally supplied we do not want to generate the csproj, as that may potentially overwrite any customization the user has performed.

Reply:
I had skimmed through the PR desc, but this makes sense - thanks for expanding on it :)